### DISCUSSION TITLE
Seeking Documentation or Guidance on KernelDB Generation

### DISCUSSION THREAD

**Discussion by NivedithaCV** [DOI-USGS/ISIS3#5329] [General]
I'm interested in learning more about the process of generating kernel databases within the project. In particular, I'm looking for information on how the kerneldbgen() function or related functionalities work.

I've gone through the code, and it seems to involve handling dependencies, creating databases for different kernel types (such as Spacecraft Clock, Leapsecond, and Extra Kernels), and utilizing the SpiceDbGen class. Is there any available resources to know how this function is used and having information about the kernel database.

**Comment by Kelvinrr** [comment]
In case you haven't seen it, the docs for the app: 

https://isis.astrogeology.usgs.gov/Application/presentation/Tabbed/kerneldbgen/kerneldbgen.html 

We have some other docs available, but they are not yet public, still going through the release process. But here is an exert that might help explain things a bit: 

> NAIF does not provide adequate resources to programmatically get all required metadata. ISIS solves this problem by creating “DB Files”, PVL files containing metadata for every kernel in the ISIS data area. These DB files are a predominately human curiated in that either the Kernel files are handwritten or a script (generally called makedb, co-located with the kernels in the kernel directory, see ISISDATA kernel folders) is handwritten to use [kerneldbgen](https://isis.astrogeology.usgs.gov/Application/presentation/Tabbed/kerneldbgen/kerneldbgen.html) app to cache most of the kernel metadata.

**Reply by NivedithaCV** [reply]
Thank you.
